lib/lbt/resources/ResourceFilterList.js

```diff
@@ -161,4 +163,24 @@ ResourceFilterList.fromString = function(filterStr) {
 	return result;
 };
 
+ResourceFilterList.negateFilters = function(patterns) {
+	return patterns.map((pattern) => {
+		let include = true;
+
+		// cut off leading '!', '-' or '+'
+		if (FILTER_PREFIXES.test(pattern)) {
+			include = pattern[0] === "+";
+			pattern = pattern.slice(1);
```

Comment:
Removing the leading character is not safe.

It's not likely that we run into problem because of that. But in case that a file name starts with !, - or +, the FILTER_PREFIXEs can be used to iavoid a misinterpretation of the filter. Remvoing the char without a replacement will break such cases.

Reply:
Could this be resolved by always adding a `+` in the else case below?

Reply:
Of course we would then need to enhance the ``filterExclude.startsWith(`${namespace}/`)`` checks in the bundlers

Reply:
> Could this be resolved by always adding a `+` in the else case below?

I think so. And yes, the checks have to b adapted then. If we rate this as a non-issue, I'm also fine with it. We just need to document it somewhere.

Reply:
So isn't this also a problem in the `makeMatcher` function from where I took the code?
It also cuts off the first character.

Should we then adopt both?

Reply:
Ok I got it. only the `negateFilters` needs to be adopted as suggested by @RandomByte

Reply:
From a discussion with Matthias:

This is not relevant for `makeMatcher` since it transform the pattern into a regular expression. At the point where it removes the first character it already decided whether the pattern is an include or exclude. Following steps don't rely on the first character.

However, if a file named `++.json` should be included, the pattern would need to be defined as `+ ++.json`. We think there is no way to improve this.

Reply:
Done in commit `ResourceFilterList.negateFilters: Always prefix includes with '+'`